### ISSUE TITLE
Faulty `let open ... in` parsing (OCaml)

### ISSUE THREAD

**Issue by nahoj** [giraud/reasonml-idea-plugin#406]
plugin version: 0.111-2023.1

### Description

I find various weird detected structures when I use `let open ... in`. For instance:

```ocaml
let is_uppercase s =
  let x = 0 in
  let open CamomileLibraryDefault.Camomile in
  let open UReStr in
  let open UReStr.Make(UTF8) in
  let y = 0 in
  let z = 0 in
  false
```

- test.ml
  - is_uppercase
    - x
  - [⬝] UReStr
  - [⬝] UReStr.Make

```ocaml
let is_uppercase s =
  let x = 0 in
  let open UReStr in
  let open UReStr.Make(UTF8) in
  let y = 0 in
  let z = 0 in
  false
```

- test.ml
  - is_uppercase
    - x
  - y
  - z

```ocaml
let is_uppercase s =
  let x = 0 in
  let open UReStr in
  let y = 0 in
  let z = 0 in
  false
```

In this one nothing is misplaced; however the `open` doesn't show:

- test.ml
  - is_uppercase
    - x
    - y
    - z

**Comment by giraud** [comment]
last one is by design. I don’t know if there is a value to show it

**Comment by giraud** [comment]
Fixed in 0.112
